argoverse/data_loading/argoverse_stereo_loader.py

```diff
@@ -0,0 +1,433 @@
+------Log id: {self.current_log}
+--------------------------------------------------------------------
+Length of the sequence: {time_in_sec:.2f} seconds
+Number of stereo pair frames (@5 Hz): {frame_image_stereo}
+        """
+
+    def __getitem__(self, key: int) -> "ArgoverseStereoLoader":
+        self.counter = key
+        self.current_log = self.log_list[self.counter]
+        self.num_disparity_frame = len(self.disparity_list)
+        self.num_stereo_camera_frame = len(self.image_timestamp_list[STEREO_FRONT_LEFT_RECT])
+        return self
+
+    def get(self, log_id: str) -> "ArgoverseStereoLoader":
+        """get ArgoverseStereoLoader object with current_log set to specified log_id
+
+        Args:
+            log_id: log id
+        Returns:
+            ArgoverseStereoLoader: with current_log set to log_id
+        """
+        self.current_log = log_id
+        self.num_stereo_camera_frame = len(self.image_timestamp_list[STEREO_FRONT_LEFT_RECT])
+        return self
+
+    def get_image_list(self, camera: str, log_id: Optional[str] = None, load: bool = False) -> List[str]:
+        """Get list of image/or image path
+
+        Args:
+            camera: camera based on camera_stats.STEREO_CAMERA_LIST
+            log_id: log_id, if not specified will use self.current_log
+            load: whether to return image array (True) or image path (False)
+
+        Returns:
+            List[str]: list of image paths (str).
+        """
+        assert self.image_list is not None
+        assert self._image_list is not None
+
+        if log_id is None:
+            log_id = self.current_log
+        if load:
+            return [self.get_image(i, camera) for i in range(len(self._image_list[log_id][camera]))]
+
+        return self._image_list[log_id][camera]
+
+    def get_disparity_list(self, name: str, log_id: Optional[str] = None, load: bool = False) -> List[str]:
+        """Get list of image/or image path
+
+        Args:
+            name: disparity map name, one of:
+
+                ["stereo_front_left_rect_disparity",
+                 "stereo_front_left_rect_objects_disparity"]
+
+            log_id: log_id, if not specified will use self.current_log
+            load: whether to return image array (True) or image path (False)
+
+        Returns:
+            List[str]: list of disparity image path (str or List[str]),
+        """
+        assert self.disparity_list is not None
+        assert self._disparity_list is not None
+
+        if log_id is None:
+            log_id = self.current_log
+        if load:
+            return [self.get_disparity_map(i, name) for i in range(len(self._disparity_list[log_id][name]))]
+
+        return self._disparity_list[log_id][name]
+
+    def get_image_at_timestamp(
+        self,
+        timestamp: int,
+        camera: str,
+        log_id: Optional[str] = None,
+        load: bool = True,
+    ) -> Optional[Union[str, np.ndarray]]:
```

Comment:
would it be cleaner to have one function that gets the image path, and another function that calls that function to get the image?

Reply:
Yeah, I think there must have better/cleaner ways to do this. I kinda followed the patterns from the argoverse_tracking_loader to be honest.